index.js

```diff
@@ -224,44 +186,103 @@ module.exports = function (config) {
       }
 
       getJobCounts(jobTypes[count])
-
     })
   }
 
-  // remove the x powered by header from all responses
-  app.use(function (req, res, next) {
-    res.removeHeader('X-Powered-By')
-    next()
-  })
+  /**
+   * app methods
+   */
 
-  // save the koop log onto the app
-  app.log = koop.log
+  /**
+   * registers koop providers, caches, and plugins
+   * @param {object} plugin - module to be registered
+   */
+  app.register = function (plugin) {
+    if (typeof plugin === 'undefined') return koop.log.error('Plugin undefined, skipping registration.')
+    if (!plugin.name) return koop.log.error('Plugin missing name, skipping registration.')
 
-  koop.Cache = new koop.DataCache(koop)
+    if (plugin.type) {
+      if (plugin.type === 'provider') return app.registerProvider(plugin)
+      if (plugin.type === 'cache') return app.registerCache(plugin)
+      if (plugin.type === 'plugin') return app.registerPlugin(plugin)
 
-  // use the default local cache until a DB adapter mod is registered
-  if (!config.db || !config.db.conn) {
-    console.warn('Warning koop w/o persistent cache means no data will be cached across server sessions.')
+      koop.log.warn('Unrecognized plugin type. Defaulting to provider.')
+      return app.registerProvider(plugin)
+    }
+
+    koop.log.warn('Plugin missing type property. Defaulting to provider.')
+    app.registerProvider(plugin)
   }
 
-  // the default cache is the local in-mem cache
-  // to persist data you must call registerCache with db adapter
-  koop.Cache.db = koop.LocalDB
+  /**
+   * registers koop providers
+   * exposes the provider's routes, controller, and model
+   * @param {object} provider - the provider to be registered
+   */
+  app.registerProvider = function (provider) {
+    app.services[provider.name] = provider
+
+    var model = new provider.model(koop) // eslint-disable-line
+    var controller = new provider.controller(model, koop.BaseController) // eslint-disable-line
```

Comment:
@kneemer and I talked about this a bit... the model and controller in providers aren't actually constructors, they're just returning an object with a bunch of methods using the module pattern. Guessing @chelm read something like http://metaduck.com/08-module-pattern-inheritance.html and decided to go for module pattern over prototypal inheritance many moons ago.

What this means is we don't need the `new` keyword here as far as I know.